### PR TITLE
languagetool: 5.1 -> 5.2

### PR DESCRIPTION
**Pull request by TredwellGit**
###### Motivation for this change
https://github.com/languagetool-org/languagetool/blob/v5.2/languagetool-standalone/CHANGES.md

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/languagetool/default.nix [hunk 1]
```diff
@@ -2,34 +2,32 @@
...
-    sha256 = "07a2cxsa04lzifphlf5mv88xpnixalmryd0blawblxsmdyhmvg3y";
+    sha256 = "1fz3rxqg5z2jxbalraz8lwkzj0jh69zzfmf3vpwywilvl7xlhdrd";
   };
   nativeBuildInputs = [ makeWrapper ];
   buildInputs = [ jre ];
 
   installPhase = ''
-    mkdir -p $out/share
-    mv * $out/share/
+    mkdir -p "$out/share"
```
Comment (SuperSandro2000) [inline pkgs/tools/text/languagetool/default.nix:15]:
```suggestion
    mkdir -p $out/share
```

File: pkgs/tools/text/languagetool/default.nix [hunk 2]
```diff
@@ -2,34 +2,32 @@
...
+    mv -- * "$out/share/"
```
Comment (SuperSandro2000) [inline pkgs/tools/text/languagetool/default.nix:16]:
```suggestion
    mv . $out/share/
```

File: pkgs/tools/text/languagetool/default.nix [hunk 3]
```diff
@@ -2,34 +2,32 @@
...
 
-    for lt in languagetool{,-commandline,-server};do
-      makeWrapper ${jre}/bin/java $out/bin/$lt \
+    for lt in languagetool{,-commandline,-server}; do
+      makeWrapper "${jre}/bin/java" "$out/bin/$lt" \
```
Comment (SuperSandro2000) [inline pkgs/tools/text/languagetool/default.nix:19]: This is not required.

File: pkgs/tools/text/languagetool/default.nix [hunk 4]
```diff
@@ -2,34 +2,32 @@
...
         --add-flags "-cp $out/share/ -jar $out/share/$lt.jar"
     done
 
-    makeWrapper ${jre}/bin/java $out/bin/languagetool-http-server \
+    makeWrapper "${jre}/bin/java" "$out/bin/languagetool-http-server" \
```
Comment (SuperSandro2000) [inline pkgs/tools/text/languagetool/default.nix:23]: This neither.

**Comment by TredwellGit** [comment]
Not required but it is better Bash.
https://github.com/koalaman/shellcheck/wiki/SC2086
https://github.com/koalaman/shellcheck/wiki/SC2035

**Comment by TredwellGit** [comment]
https://github.com/NixOS/nixpkgs/pull/105641

**Comment by SuperSandro2000** [comment]
> #105641

That is for sh files which are often writte in sh. If we would add " to packages we would need to change all packages specifying any phase, pre or post.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108912` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>languagetool</li>
<li>vimPlugins.lens-vim</li>
<li>vimPlugins.vim-grammarous</li>
</ul>
</details>

**Pull request merged**